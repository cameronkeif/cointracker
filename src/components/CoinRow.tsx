import React from 'react';
import { Sparklines, SparklinesLine } from 'react-sparklines';
import { red } from '@material-ui/core/colors';
import RemoveIcon from '@material-ui/icons/RemoveCircle';

import { CoinItem } from '../react-app-env';

export type CoinRowProps = {
  coin: CoinItem,
  onRemove: Function,
};

/**
 * A row in the coin table, takes in a coin and outputs the necessary table row
 */
const CoinRow = ({ coin, onRemove }: CoinRowProps) => (
  <tr className="coin-row">
    <td>
      <img className="coin-image" src={coin.iconUrl} alt={coin.name} />
      {coin.name}
    </td>
    <td>{coin.symbol}</td>
    <td>
      {
      parseFloat(coin.price).toLocaleString(
        undefined,
        {
          maximumFractionDigits: 2,
          minimumFractionDigits: 2,
        },
      )
}

    </td>
    <td className={coin.change >= 0 ? 'positive-change' : 'negative-change'}>
      {coin.change}
      %
    </td>
    <td>
      <Sparklines data={coin.history} svgWidth={100} svgHeight={26} margin={5}>
        <SparklinesLine color={coin.change >= 0 ? 'green' : 'red'} />
      </Sparklines>
    </td>
    <td>
      <RemoveIcon
        style={{ color: red[500], cursor: 'pointer' }}
        fontSize="small"
        onClick={() => onRemove(coin.symbol)}
      />
    </td>
  </tr>
);

export default CoinRow;
