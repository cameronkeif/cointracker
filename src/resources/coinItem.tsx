import { CoinItem } from '../react-app-env';

const coinItem: CoinItem = {
  change: -2.06,
  symbol: 'BTC',
  name: 'Bitcoin',
  uuid: 'Qwsogvtv82FCd',
  price: '55099.8441599246',
  iconUrl: 'https://cdn.coinranking.com/bOabBYkcX/bitcoin_btc.svg',
};

export default coinItem;
