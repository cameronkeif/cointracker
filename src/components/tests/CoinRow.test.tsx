import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CoinRow, { CoinRowProps } from '../CoinRow';
import coinItem from '../../resources/coinItem';

const defaultProps: CoinRowProps = {
  coin: coinItem,
  onRemove: jest.fn(),
};

describe('CoinRow.tsx', () => {
  it('renders correctly (positive change)', () => {
    const { asFragment } = render(
      <CoinRow
        {...defaultProps}
        coin={{ ...coinItem, change: 1.52 }}
      />,
    );
    expect(asFragment).toMatchSnapshot();
  });

  it('renders correctly (negative change)', () => {
    const { asFragment } = render(
      <CoinRow
        {...defaultProps}
      />,
    );
    expect(asFragment).toMatchSnapshot();
  });

  it('calls onRemove when the remove button is clicked', () => {
    const onRemove = jest.fn();
    expect(onRemove.mock.calls.length).toBe(0);

    render(
      <CoinRow
        {...defaultProps}
        onRemove={onRemove}
      />,
    );
    fireEvent.click(screen.getByTitle('Remove'));
    expect(onRemove).toHaveBeenCalledWith(coinItem.symbol);
  });
});
