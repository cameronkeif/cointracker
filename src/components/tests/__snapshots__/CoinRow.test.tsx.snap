// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinRow.tsx renders correctly (negative change) 1`] = `[Function]`;

exports[`CoinRow.tsx renders correctly (positive change) 1`] = `[Function]`;
